app/src/main/java/org/simple/clinic/AppDatabase.kt

```diff
@@ -164,6 +166,9 @@ abstract class AppDatabase : RoomDatabase() {
 
   abstract fun teleconsultRecordDao(): TeleconsultRecord.RoomDao
 
+  @Inject
+  lateinit var userClock: UserClock
```

Comment:
Ideally, we should not inject it into this `AppDatabase` abstract class, instead, we can pass the `UserClock` to the `prune` from the call site. You can make that change in later PR.

Reply:
I am amazed that this didn't throw in dagger build errors 🤔

Reply:
Ideally, the `prune` method should just take an `Instant` as a parameter. Not even the clock directly.

Reply:
+1